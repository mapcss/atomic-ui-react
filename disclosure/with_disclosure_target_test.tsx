import WithDisclosureTarget from "./with_disclosure_target.ts";
import Disclosure from "./disclosure.ts";
import SSRProvider from "../ssr/ssr_provider.ts";
import {
  assertSnapshot,
  describe,
  expect,
  it,
  setupJSDOM,
} from "../dev_deps.ts";
import { render } from "@testing-library/react";

const describeTests = describe({
  name: "WithDisclosureTarget",
  async beforeEach() {
    await setupJSDOM();
  },
});

it(describeTests, "should throw error when it not wrap Context", () => {
  expect(() =>
    render(
      <WithDisclosureTarget>
        <p>test</p>
      </WithDisclosureTarget>,
    )
  ).toThrow();
});

it(
  describeTests,
  "should render with style when the internal `isOpen` is false",
  (t) => {
    const { baseElement } = render(
      <WithDisclosureTarget>
        <p>test</p>
      </WithDisclosureTarget>,
      {
        wrapper: ({ children }) => {
          return (
            <SSRProvider>
              <Disclosure>{children as never}</Disclosure>
            </SSRProvider>
          );
        },
      },
    );

    assertSnapshot(t, baseElement.innerHTML);
  },
);

it(describeTests, "should render when the internal `isOpen` is true", (t) => {
  const { baseElement } = render(
    <WithDisclosureTarget>
      <p>test</p>
    </WithDisclosureTarget>,
    {
      wrapper: ({ children }) => {
        return (
          <SSRProvider>
            <Disclosure isDefaultOpen>{children as never}</Disclosure>
          </SSRProvider>
        );
      },
    },
  );

  assertSnapshot(t, baseElement.innerHTML);
});

it(describeTests, "should overwrite default closed style", (t) => {
  const { baseElement } = render(
    <WithDisclosureTarget
      closedStyle={{
        visibility: "hidden",
      }}
    >
      <p>test</p>
    </WithDisclosureTarget>,
    {
      wrapper: ({ children }) => {
        return (
          <SSRProvider>
            <Disclosure>{children as never}</Disclosure>
          </SSRProvider>
        );
      },
    },
  );

  assertSnapshot(t, baseElement.innerHTML);
});

it(describeTests, "should merge style", (t) => {
  const { baseElement } = render(
    <WithDisclosureTarget>
      <p style={{ font: "icon" }}>test</p>
    </WithDisclosureTarget>,
    {
      wrapper: ({ children }) => {
        return (
          <SSRProvider>
            <Disclosure>{children as never}</Disclosure>
          </SSRProvider>
        );
      },
    },
  );

  assertSnapshot(t, baseElement.innerHTML);
});
