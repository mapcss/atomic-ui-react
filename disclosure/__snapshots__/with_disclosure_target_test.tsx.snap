export const snapshot = {};

snapshot[`WithDisclosureTarget > should render with style when the internal \`isOpen\` is false 1`] = `'<div><p id="atomic-ui-0" style="display: none;">test</p></div>'`;

snapshot[`WithDisclosureTarget > should render when the internal \`isOpen\` is true 1`] = `'<div><p id="atomic-ui-0">test</p></div>'`;

snapshot[`WithDisclosureTarget > should overwrite default closed style 1`] = `'<div><p id="atomic-ui-0" style="visibility: hidden;">test</p></div>'`;

snapshot[`WithDisclosureTarget > should merge style 1`] = `'<div><p style="font: icon; display: none;" id="atomic-ui-0">test</p></div>'`;
